pkg/auth/handler/webapp/authflow_forgot_password.go

```diff
@@ -96,16 +113,35 @@ func (h *AuthflowForgotPasswordHandler) GetData(w http.ResponseWriter, r *http.R
 func (h *AuthflowForgotPasswordHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	flowName := "default"
 	var handlers AuthflowControllerHandlers
-	handlers.Get(func(s *webapp.Session, screen *webapp.AuthflowScreenWithFlowResponse) error {
 
-		data, err := h.GetData(w, r, s, screen)
+	handlers.Get(func(s *webapp.Session, screen1 *webapp.AuthflowScreenWithFlowResponse) error {
+
+		var screen2 *webapp.AuthflowScreenWithFlowResponse
+
+		identification := r.Form.Get("q_login_id_input_type")
+		loginID := r.Form.Get("q_login_id")
+
+		if identification != "" && loginID != "" {
+			var err error
+			// Advance one step to get the options
+			_, screen2, err = h.Controller.AdvanceWithInput(r, s, screen1, map[string]interface{}{
```

Comment:
I think this should be handled by

1. Not to use HandleStartOfFlow, use something else that support creating a flow AND pass an input at the same time. Then we do not need to call AdvanceWithInput in GET. We also need not to change the signature of AdvanceWithInput.

Reply:
@louischan-oursky Updated to add a optional input to `HandleStartOfFlow`, see if it is better?